src/Span.ts

```diff
@@ -423,8 +440,71 @@ const getDefaults = (type: EventType): IDefaultActions => {
   }
 }
 
+const setHttpHeader = (context: TypeSpanContext, type: HttpRequestOptions, headers: { [key: string]: any }): { [key: string]: any } => {
+
+  const { traceId, parentSpanId, spanId, flags, sampled } = context
+
+  switch (type) {
+
+    case HttpRequestOptions.xb3: {
+      let XB3headers = {
+        'X-B3-TraceId': traceId,
+        'X-B3-SpanId': spanId,
+        'X-B3-Sampled': sampled,
+        'X-B3-Flags': flags,
+        'X-B3-Version': '0'
+      }
+      let result = parentSpanId ? Object.assign({ 'X-B3-ParentSpanId': parentSpanId }, XB3headers) : XB3headers
+      return Object.assign(headers, result)
+    }
+
+    case HttpRequestOptions.w3c:
+    default: {
+      const version = Buffer.alloc(1).fill(0)
+      const flagsForBuff = (flags && sampled) ? (flags | sampled) : flags ? flags : sampled ? sampled : 0x00
+      const flagsBuffer = Buffer.alloc(1).fill(flagsForBuff)
+      const traceIdBuff = Buffer.from(traceId, 'hex')
+      const spanIdBuff = Buffer.from(spanId, 'hex')
+      const parentSpanIdBuff = parentSpanId && Buffer.from(parentSpanId, 'hex')
+      let result = {}
+      let W3CHeaders = parentSpanIdBuff
+        ? new TraceParent(Buffer.concat([version, traceIdBuff, spanIdBuff, flagsBuffer, parentSpanIdBuff]))
+        : new TraceParent(Buffer.concat([version, traceIdBuff, spanIdBuff, flagsBuffer]))
+      if (headers.tracestate) {
+        return Object.assign({ traceparent: W3CHeaders.toString() }, { tracestate: createTracestate(headers.tracestate, traceId), headers })
+      }
+      return Object.assign({ traceparent: W3CHeaders.toString() }, headers)
+    }
+  }
+}
+
+const createTracestate = (tracestate: string, opaqueValue: string): string => {
```

Comment:
please move this into the corresponding context, and or rename it to be more specific (i.e. createTracestate --> createTraceW3cState)